src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionMetadataProvider.cs

```diff
@@ -102,6 +104,12 @@ public static class ExpressionMetadataProvider
                 //    m => m.Color.Red (nested property access)
                 //    m => m.Widgets[0].Size (expression ending with property-access)
                 metadata = metadataProvider.GetMetadataForType(containerType).Properties[propertyName];
+                if (metadata == null)
+                {
+                    throw new InvalidOperationException(Resources.FormatModelExpression_InvalidProperty(
+                        nameof(ModelMetadata),
+                        containerType.Name + "." + propertyName));
```

Comment:
Talked offline: @rynowak hadn't seen the resource I suggested passing the separate property and type names into when he commented on this.